integration/api_endpoints_test.go

```diff
@@ -83,3 +83,19 @@ func TestConfigAPIEndpoint(t *testing.T) {
 	mimir2 := e2emimir.NewSingleBinary("mimir-2", nil, e2emimir.WithPorts(9009, 9095), e2emimir.WithConfigFile(mimirConfigFile))
 	require.NoError(t, s.StartAndWaitReady(mimir2))
 }
+
+func TestFormatQueryAPIEndpoint(t *testing.T) {
+	// Start Mimir in single binary mode, reading the config from file
+	s, mimir1 := newMimirSingleBinaryWithLocalFilesytemBucket(t, "mimir-1", nil)
+	defer s.Close()
+	require.NoError(t, s.StartAndWaitReady(mimir1))
+
+	// Get config from /prometheus/api/v1/format_query API endpoint.
+	res, err := e2e.DoGet(fmt.Sprintf("http://%s/prometheus/api/v1/format_query?query=count(up)by(foo)", mimir1.Endpoint(9009)))
+	require.NoError(t, err)
+
+	defer runutil.ExhaustCloseWithErrCapture(&err, res.Body, "config API response")
+	_, err = io.ReadAll(res.Body)
```

Comment:
Should we make an assertion on the contents of the response?

Reply:
right, done https://github.com/grafana/mimir/pull/4373/commits/58d017091f77e794d64bb1729e4b3ec23177836b